factory/alchemy.py

```diff
@@ -1,10 +1,14 @@
 # Copyright: See the LICENSE file.
 
+from typing import TypeVar
+
 from sqlalchemy.exc import IntegrityError
 from sqlalchemy.orm.exc import NoResultFound
 
 from . import base, errors
 
+T = TypeVar("T")
```

Comment:
Is it required to create a new TypeVar here? Can't we import it from `base.py` or user `base.T`? The same applies to`django.py` and `mongo.py`.